### PR TITLE
fix(components): [table] fix cell tooltip display error

### LINKED ISSUE #15004: [Component] [table] el-table    show-overflow-tooltip  在body  transform:scale(1.33333)的情况下，位置偏移，并且tooltip 显示错误
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.25`
- Element Plus Version: `2.2.36`
- Browser / OS: `Chrome 110.0.5481.178（正式版本） （64 位）`
- Build Tool: `Vite`

### Related Component

- `el-table`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6IjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG5pbXBvcnQgeyByZWYsIHZlcnNpb24gYXMgdnVlVmVyc2lvbiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IHZlcnNpb24gYXMgZXBWZXJzaW9uIH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuaW1wb3J0IHsgRWxlbWVudFBsdXMgfSBmcm9tICdAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZSdcblxuY29uc3QgbXNnID0gcmVmKCdIZWxsbyBXb3JsZCEnKVxuXG5sZXQgbGlzdDEgPSByZWYoW1xue1xuIFwibmFtZVwiOiBcIuW8oOS4gFwiLFxuICAgICAgICAgICAgXCJnZW5kZXJcIjogXCLnlLdcIixcbiAgICAgICAgICAgIFwibnVtYmVyXCI6IFwiMVwiLFxuICAgICAgICAgICAgXCJyZW1hcmtcIjogXCJYWFhYXCIsXG4gICAgICAgICAgICBcIui6q+S7vVwiOiBcIuiuv+WuolwiLFxuICAgICAgICAgICAgXCJpZGVudGl0eUNhcmRcIjogXCIxMTExMTExMTExMTExMTExMTFcIixcbn1cbl0pXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8ZGl2IHN0eWxlPVwidHJhbnNpdGlvbjogbm9uZXMgMHMgZWFzZSAwcztcbiAgICB0cmFuc2Zvcm06IHNjYWxlKDEuMzMzMzMpO21hcmdpbi1sZWZ0OjIwMHB4O1wiPlxuICA8ZWwtdGFibGVcbiAgICAgICAgICAgICAgXG4gICAgICAgICAgICAgICAgc2hvdy1vdmVyZmxvdy10b29sdGlwXG4gICAgICAgICAgICAgICAgOmRhdGE9XCJsaXN0MVwiXG4gICAgICAgICAgICAgICAgc3R5bGU9XCJ3aWR0aDogNDAwcHhcIlxuICAgICAgICAgICAgICAgIHNpemU9XCJzbWFsbFwiXG4gICAgICAgICAgICA+XG4gICAgICAgICAgICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwibnVtYmVyXCIgbGFiZWw9XCLnvJblj7dcIiB3aWR0aD1cIjQ2cHhcIiAvPlxuICAgICAgICAgICAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cIm5hbWVcIiBsYWJlbD1cIuWnk+WQjVwiIHdpZHRoPVwiNTRweFwiIC8+XG4gICAgICAgICAgICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwi6Lqr5Lu9XCIgbGFiZWw9XCLouqvku71cIiB3aWR0aD1cIjQwcHhcIiAvPlxuICAgICAgICAgICAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cImlkZW50aXR5Q2FyZFwiIGxhYmVsPVwi6K+B5Lu25Y+3XCIgd2lkdGg9XCIxMDRweFwiIC8+XG4gICAgICAgICAgICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwiZ2VuZGVyXCIgbGFiZWw9XCLmgKfliKtcIiB3aWR0aD1cIjQwcHhcIiAvPlxuICAgICAgICAgICAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cInJlbWFya1wiIGxhYmVsPVwi5aSH5rOoXCIgd2lkdGg9XCI0NHB4XCI+XG4gICAgICAgICAgICAgICAgICAgXG4gICAgICAgICAgICAgICAgPC9lbC10YWJsZS1jb2x1bW4+XG4gICAgICAgICAgICA8L2VsLXRhYmxlPlxuICA8L2Rpdj5cbiBcbiAgPHA+XG4gICAgPGVsLWljb24gY29sb3I9XCJ2YXIoLS1lbC1jb2xvci1wcmltYXJ5KVwiPjxFbGVtZW50UGx1cyAvPjwvZWwtaWNvbj5cbiAgICBFbGVtZW50IFBsdXMge3sgZXBWZXJzaW9uIH19ICsgVnVlIHt7IHZ1ZVZlcnNpb24gfX1cbiAgPC9wPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIl9vIjp7fX0=)

### Steps to reproduce

1.在el-table的上级或者上上级加 transform:scale
2.鼠标放在td上，没有超出隐藏的也会鼠标放上去出现tooltip

### What is Expected?

没有超出隐藏鼠标放上去不会出现tooltip

### What is actually happening?

没有超出隐藏的也会鼠标放上去出现tooltip

### Additional comments

只要是加了缩放，element-plus 组件pop位置都会不正确，希望能够识别到缩放，或者加一个缩放参数，算出准确的位置跟宽度

### PR DESCRIPTION
**Pull request by electroluxcode**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ x] Make sure you are merging your commits to `dev` branch.
- [ x] Add some descriptions and refer to relative issues for your PR.

fix #15004

修复trigger 到 table cell 的触发方式，原来的代码中 cellChild 的计算方式是通过 offsetWidth 和 offsetHeight(render前的元素)，但是在后面根据条件触发是用 getComputeStyle 和 getBoundingClientRect 这个api(render后的元素)和render前的cellChild进行比较 。这会导致下面链接呈现的情况

- https://codesandbox.io/p/sandbox/element-plus-vue3-yzcky9?file=%2Fsrc%2FApp.vue%3A13%2C23   (设置了show-overflow-tooltip的情况下，在数据量不多的时候,例如hover链接里面的N，仍然能够触发tooltip)

solution: 用 getBoundingClientRect 替代 offsetwidth/height

### CODE REVIEW

**Comment by electroluxcode** [comment]
- before
   - https://codesandbox.io/p/sandbox/element-plus-vue3-yzcky9?file=%2Fsrc%2FApp.vue%3A13%2C23 
- after
  - https://codesandbox.io/p/github/electroluxcode/element-plus/fix-trigger/demo?workspaceId=137471a6-f9d2-4b79-bfb2-49bc9d3932ec
    - nvm install 18.14.1
    - pnpm install
    - go to url "/en-US/component/table.html"


Could anyone help me with a code review?

**Comment by electroluxcode** [comment]
related issue: https://github.com/element-plus/element-plus/issues/15004

File: packages/components/table/src/table-body/events-helper.ts [hunk 1]
```diff
@@ -134,6 +134,7 @@ function useEvents<T>(props: Partial<TableBodyProps<T>>) {
     let rangeWidth = range.getBoundingClientRect().width
     let rangeHeight = range.getBoundingClientRect().height
     const offsetWidth = rangeWidth - Math.floor(rangeWidth)
+    const { width: cellChildWidth, height: cellChildHeight } = cellChild.getBoundingClientRect()
```
Comment (btea) [inline packages/components/table/src/table-body/events-helper.ts:137]:
```suggestion
    const { width: cellChildWidth, height: cellChildHeight } =
      cellChild.getBoundingClientRect()
```

**Review by btea — APPROVED** [review]
LGTM, Thanks!

**Review by kooriookami — APPROVED** [review]

**Pull request merged**